controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -577,30 +568,111 @@ func setAvailableCondition(_ context.Context, kcp *controlplanev1.KubeadmControl
 			controlplanev1.KubeadmControlPlaneMachineEtcdMemberHealthyV1Beta2Condition,
 			controlplanev1.KubeadmControlPlaneMachineEtcdPodHealthyV1Beta2Condition) {
 			k8sControlPlaneNotHealthy++
+		} else {
+			k8sControlPlaneNotHealthyButNotReportedYet++
 		}
 	}
 
+	// Maps machines to members
+	memberToMachineMap := map[string]*clusterv1.Machine{}
+	provisioningMachines := []*clusterv1.Machine{}
+	for _, machine := range machines {
+		if machine.Status.NodeRef == nil {
+			provisioningMachines = append(provisioningMachines, machine)
+			continue
+		}
+		for _, member := range etcdMembers {
+			if machine.Status.NodeRef.Name == member.Name {
+				memberToMachineMap[member.Name] = machine
+				break
+			}
+		}
+	}
+
```

Comment:
What about `if etcdIsManaged`?

Reply:
We are reporting status for both etcd and control plane components in case etcd is managed
In case etcd is NOT managed we report only about control plane components

Reply:
I will make this more explicit, but the TLDR is that everything just works because etcdmember list is empty when etcd is not managed